services/streammanager/wunderkind/wunderkindmanager_test.go

```diff
@@ -0,0 +1,147 @@
+package wunderkind
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/rudderlabs/rudder-go-kit/logger/mock_logger"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	mock_lambda "github.com/rudderlabs/rudder-server/mocks/services/streammanager/lambda"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+var (
+	sampleMessage    = "sample payload"
+	sampleFunction   = "sampleLambdaFunction"
+	sampleExternalID = "sampleExternalID"
+	sampleIAMRoleARN = "sampleRoleArn"
+	invocationType   = "RequestResponse"
+)
+
+func TestNewProducer(t *testing.T) {
+	t.Setenv("WUNDERKIND_REGION", "us-east-1")
+	t.Setenv("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
+	t.Setenv("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
+	t.Setenv("WUNDERKIND_LAMBDA", sampleFunction)
```

Comment:
t Setenv should be avoided when better alternatives are possible. We should pass a config struct instead

Reply:
NewProducer does not accept config struct in function parameter.

Reply:
You can pass *config.Config as DI and avoid global invocation. 
```GO
conf := config.New()
	conf.Set("WUNDERKIND_REGION", "us-east-1")
	conf.Set("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
	conf.Set("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
	conf.Set("WUNDERKIND_LAMBDA", sampleFunction)
```

Reply:
addressed